cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -413,12 +482,57 @@ pub mod tests {
             check_build_metadata(&env, EXAMPLE_PACKAGE_NAME, &flox.system).unwrap(),
         );
 
-        let publish_provider = PublishProvider::<&FloxBuildMk> {
+        let publish_provider = PublishProvider::<&FloxBuildMk, &MockCache> {
             build_metadata,
             env_metadata,
+            cache: None,
             _builder: None,
         };
 
-        let _res = publish_provider.publish(&client, &catalog_name);
+        let _res = publish_provider.publish(&client, &catalog_name).await;
+
+        assert!(_res.is_ok());
```

Comment:
how did this test anything before if we only returned a future, but we never executed/awaited it?
i guess it didn't..